### PR TITLE
chore(hybridcloud) Remove SnowflakeIdMixin

### PR DESCRIPTION
**Pull request by markstory**
This mixin complicates typing and can be implemented as a decorator and module functions.

Refs HC-1192

### CODE REVIEW

File: src/sentry/utils/snowflake.py [hunk 1]
```diff
@@ -15,44 +16,52 @@
...
+    Check if a model class has been annotated as a snowflake id model
+    """
+    name = f"{model_class.__module__}.{model_class.__name__}"
+    return name in _snowflake_models
+
+
+def save_with_snowflake_id(
+    instance: BaseModel, snowflake_redis_key: str, save_callback: Callable[[], object]
+) -> None:
+    for _ in range(settings.MAX_REDIS_SNOWFLAKE_RETRY_COUNTER):
```
Comment (corps) [inline src/sentry/utils/snowflake.py:48]: Should this assert that the model in question `uses_snowflake_id`?

**Review by corps — APPROVED** [review]

**Review by asottile-sentry — COMMENTED** [review]
nice! much better

File: src/sentry/utils/snowflake.py [hunk 2]
```diff
@@ -15,44 +16,56 @@
...
 if TYPE_CHECKING:
-    from sentry.db.models.base import Model
+    from sentry.db.models.base import Model as BaseModel
 
 _TTL = timedelta(minutes=5)
 
+_snowflake_models: set[str] = set()
+
+
+def snowflake_id_model(model_class: type[Model]) -> object:
```
Comment (asottile-sentry) [inline src/sentry/utils/snowflake.py:26]: this should be

```python
ModelT = TypeVar("ModelT", bound=Model)

def snowflake_id_model(model_class: type[ModelT]) -> type[ModelT]:
   ...
```

that way the decorator preserves the type

Comment (asottile-sentry) [reply src/sentry/utils/snowflake.py:26]: github is acting very strangely -- this comment might post twice

**Review by dashed — APPROVED** [review]

**Pull request merged**